web/src/main/java/org/openmrs/web/controller/form/FormResourceController.java

```diff
@@ -81,7 +84,14 @@ public String handleAddFormResource(@ModelAttribute("resource") FormResource res
 		if (errors.hasErrors()) {
 			throw new RuntimeException("Error handling not yet implemented");
 		} else {
-			Context.getFormService().saveFormResource(resource);
+			try {
+				Context.getFormService().saveFormResource(resource);
+			}
+			catch (Exception ex) {
+				HttpSession httpSession = request.getSession();
+				Log.error("Error while adding form resource", ex);
+				httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "form.resource.add.failure");
```

Comment:
Is there a way that we can give a more specific error message? (E.g. in the specific case reported in this ticket, someone tried to upload a binary file when only text is accepted. Is there any way we can tell that's the error and report it?)